modules/core/image/src/Plugin/ImageEffect/AutoOrientImageEffect.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\farm_image\Plugin\ImageEffect;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Image\ImageInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\image\Attribute\ImageEffect;
+use Drupal\image\ConfigurableImageEffectBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Automatically rotates an image based on EXIF Orientation.
+ */
+#[ImageEffect(
+  id: "farm_image_auto_orient",
+  label: new TranslatableMarkup("Auto-orient"),
+  description: new TranslatableMarkup("Automatically rotates an image based on EXIF Orientation.")
+)]
+class AutoOrientImageEffect extends ConfigurableImageEffectBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * AutoOrientImageEffect constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   The file system service.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, LoggerInterface $logger, FileSystemInterface $file_system) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $logger);
+    $this->fileSystem = $file_system;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory')->get('image'),
+      $container->get('file_system'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applyEffect(ImageInterface $image) {
+
+    // Only proceed if the exif PHP extension is loaded.
+    if (!extension_loaded('exif')) {
+      return TRUE;
+    }
+
+    // Only work with supported image types.
+    $mime_types = [
+      'image/jpeg',
+      'image/png',
+      'image/tiff',
+    ];
```

Comment:
Just noting that `tiff` isn't included in our `image` field file extensions: https://github.com/farmOS/farmOS/blob/b99e1ae519bf79c8e28ca76ab922d30a13d20427/modules/core/field/src/FarmFieldFactory.php#L631-L632

I read that `gif` doesn't include exif metadata so that's OK. Also wondering if we should add support for `webp` images, but could be a followup.

Reply:
That's true. Perhaps we should add `tiff` as an option in our allowed extensions.

I included `tiff` in the allowed mime types because TIFFs support EXIF data. But you're right, `webp` also supports EXIF, so it should be included in the allowed mime types too.

And both should be added to our allowed file extensions in image fields, IMO.

Reply:
Added a commit to this PR: https://github.com/farmOS/farmOS/pull/941/commits/8dbb9a87c3690f5f2b4574dfcf8dec4c7d18bd60

> And both should be added to our allowed file extensions in image fields, IMO.

I will open a separate PR for this.

Reply:
Followup PR: https://github.com/farmOS/farmOS/pull/942